### DISCUSSION TITLE
Service Install Succeeds but throws error and fails main installer (halts when using `/S`)

### DISCUSSION THREAD

**Discussion by Vacant0mens** [rabbitmq/rabbitmq-server#12136] [Questions]
### Describe the bug

When running the rabbitmq-server-3.12.6.exe installer, the service creation fails on a clean install of Windows Server 2019.

Running the installer with the GUI shows a message box that says "rabbitmq-service.bat install failed with exit code 1" and cancels the install, even though everything else was successful, and the install runs just fine with the "Rabbit Service" option un-checked.

When running the `rabbitmq-service.bat install`, it throws:
`C:\Program Files\Erlang OTP\erts-14.0.2\bin\erlsrv: Unrecognized option {child_type,worker}].`

Despite the error, the service was still installed and starts successfully. This error shows after the message that says "Service RabbitMQ added to System."

### Reproduction steps

1. Fresh install of Windows Server 2019
2. Install Erlang 26
3. run `rabbitmq-server-3.12.6.exe` as administrator
4. go through install wizard with all default options
5. install
6. progress bar gets close to the end, but then a message box pops up saying "rabbitmq-service.bat install failed with exit code 1"
7. open command line as administrator
8. `cd` to rabbitmq's `sbin` folder
9. run `rabbitmq-service.bat install`
10. output shows success message, and then shows: `C:\Program Files\Erlang OTP\erts-14.0.2\bin\erlsrv: Unrecognized option {child_type,worker}].`

after this, running `rabbitmq-service.bat start` also succeeds.

### Expected behavior

successful install of RabbitMQ Server and Service

### Additional context

the silent install with `/S` also fails and pops up the message box. However, running the chocolatey install remotely hangs since it can't close the message box.

**Comment by lukebakken** [comment]
Hello,

I just fired up a Windows 2019 VM using this `Vagrantfile`

```
Vagrant.configure("2") do |config|
  config.vm.box = "gusztavvargadr/windows-server"
  config.vm.box_version = "1809.0.2208"
  config.vm.synced_folder ".", "/vagrant", disabled: true
end
```

I installed Erlang 26.2.5.2 from here:

https://github.com/erlang/otp/releases/download/OTP-26.2.5.2/otp_win64_26.2.5.2.exe

...and the latest RabbitMQ from here:

https://github.com/rabbitmq/rabbitmq-server/releases/download/v3.13.7/rabbitmq-server-3.13.7.exe

Everything worked as expected. RabbitMQ 3.12.x does not receive community support from Team RabbitMQ.

**Reply by Vacant0mens** [reply]
Sorry. I should have mentioned that I was installing Erlang 26.0.2
Not sure if that makes a difference or not.

**Comment by michaelklishin** [comment]
@lukebakken the point may be that if the service fails, so will the package installation. An option like that sounds reasonable to me but this entire package is OSS, and this has not been requested in its ≈ 8 year long history, so @Vacant0mens should contribute this change.

It is a classic case of "open source is for being able to scratch your own itch."

**Reply by lukebakken** [reply]
>  if the service fails, so will the package installation

It seems that's not what @Vacant0mens is reporting. In this case, the files are all copied, so that after the installer is exited (due to a service installation error), you _can_ install the service from the command link. I have no idea why `Unrecognized option {child_type,worker}]` is the error here, but I have demonstrated, with a fresh Win 2019 environment, that this does not happen using the latest RabbitMQ release. So, 🤷
